src/radix_connect/mobile/deep_link_parsing/request.rs

```diff
@@ -0,0 +1,162 @@
+use crypto::signatures::ed25519::Signature;
+
+use super::super::session::session_id::SessionID;
+use crate::prelude::*;
+use hex::ToHex;
+
+#[derive(Debug, PartialEq)]
+pub struct RadixConnectMobileRequest {
+    pub session_id: SessionID,
+    pub origin: DappOrigin,
+    pub public_key: KeyAgreementPublicKey,
+    pub identity_public_key: Ed25519PublicKey,
+    pub dapp_definition_address: DappDefinitionAddress,
+    pub signature: Ed25519Signature,
+    pub interaction: DappToWalletInteractionUnvalidated,
+}
+
+impl RadixConnectMobileRequest {
+    pub fn new(
+        session_id: SessionID,
+        origin: DappOrigin,
+        public_key: KeyAgreementPublicKey,
+        identity_public_key: Ed25519PublicKey,
+        dapp_definition_address: DappDefinitionAddress,
+        signature: Ed25519Signature,
+        interaction: DappToWalletInteractionUnvalidated,
+    ) -> Self {
+        Self {
+            session_id,
+            origin,
+            public_key,
+            identity_public_key,
+            dapp_definition_address,
+            signature,
+            interaction,
+        }
+    }
+}
+
+impl HasSampleValues for RadixConnectMobileRequest {
+    fn sample() -> Self {
+        RadixConnectMobileRequest::new(
+            SessionID::sample(),
+            DappOrigin::sample(),
+            KeyAgreementPublicKey::sample(),
+            Ed25519PublicKey::sample(),
+            DappDefinitionAddress::sample(),
+            Ed25519Signature::sample(),
+            DappToWalletInteractionUnvalidated::sample(),
+        )
+    }
+
+    fn sample_other() -> Self {
+        RadixConnectMobileRequest::new(
```

Comment:
I typically use `sample_other` for all values inside a higher level `sample_other` (if I dont have good reason as to not doing so, maybe you have that too?)